src/Runner.php

```diff
@@ -719,54 +716,51 @@ private function processChildProcs($childProcs)
         $success = true;
 
         while (count($childProcs) > 0) {
-            foreach ($childProcs as $key => $procData) {
-                $res = pcntl_waitpid($procData['pid'], $status, WNOHANG);
-                if ($res === $procData['pid']) {
-                    if (file_exists($procData['out']) === true) {
-                        include $procData['out'];
-
-                        unlink($procData['out']);
-                        unset($childProcs[$key]);
-
-                        $numProcessed++;
-
-                        if (isset($childOutput) === false) {
-                            // The child process died, so the run has failed.
-                            $file = new DummyFile('', $this->ruleset, $this->config);
-                            $file->setErrorCounts(1, 0, 0, 0);
-                            $this->printProgress($file, $totalBatches, $numProcessed);
-                            $success = false;
-                            continue;
-                        }
+            $pid = pcntl_waitpid(0, $status);
+            $out = $childProcs[$pid];
```

Comment:
Should at least run an `isset()` here in case `pcntl_waitpid` returns an error code and not a PID we know about.

Reply:
Thanks. I didn't notice the possible error in the return value.

 I believe instead of checking for `isset()`, it would make more sense to check for the error and assume that the returned PID is always known to the runner. At lest if it happens to be not true, we'll see a notice instead of this issue silently ignored.